server/models.py

```diff
@@ -47,18 +57,20 @@ def __repr__(self):
         return '<User %r>' % self.email
 
     # TODO: Cache enrollment queries
-    def enrollments(self, roles=['student']):
-        return Participant.query.filter(
-            Participant.user == self.id,
-            Participant.role.in_(roles)
-        ).all()
+    def enrollments(self, roles=[STUDENT_ROLE]):
+        return [e for e in self.participations if e.role in roles]
+
+    def is_enrolled(self, course_id, roles=VALID_ROLES):
+        for enroll in self.participations:
```

Comment:
Should be a db filter, not a Python filter - `self.participations` will get really large